File: Dev/Swift/Sources/SKAISwiftUI/PlaceholderTab.swift
import SwiftUI

/// One tab in a tabbed placeholder scene.
///
/// Each tab hosts a child scene's placeholder, so click-through exercises real
/// tab switching rather than a faked stack.
public struct PlaceholderTab: Identifiable {
    public let id = UUID()

    /// Tab label — the child scene's name.
    public let label: String

    /// SF Symbol name for the tab item. Use a base name with a `.fill` variant
    /// so iOS solidifies it when selected. If it doesn't resolve, the scene
    /// falls back to a default.
    public let systemImage: String

    /// The tab's hosted child-scene placeholder (type-erased for array storage).
    public let content: AnyView

    public init(label: String, systemImage: String = "square", @ViewBuilder content: () -> some View) {
        self.label = label
        self.systemImage = systemImage
        self.content = AnyView(content())
    }
}

File: Dev/Compose/skai-compose-android/src/main/java/com/swiftkickmobile/skai/compose/placeholder/PlaceholderTab.kt
package com.swiftkickmobile.skai.compose.placeholder

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Star
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.vector.ImageVector
import com.swiftkickmobile.skai.compose.navigation.Route

/**
 * One tab in a tabbed placeholder scene.
 *
 * Each tab is a destination in the scene's tab `NavHost`, switched with
 * `navigateToTab`, so click-through exercises real tab navigation (per-tab
 * state saved and restored, back from any tab root returning to the primary
 * tab) rather than a faked stack.
 *
 * @param label Tab label — the child scene's name.
 * @param route The tab's typed route case. The first tab's route is the primary
 *   tab and the tab host's start destination.
 * @param icon Icon for the tab item; defaults to a generic placeholder glyph.
 * @param content The tab's hosted child-scene placeholder.
 */
data class PlaceholderTab(
    val label: String,
    val route: Route,
    val icon: ImageVector = Icons.Default.Star,
    val content: @Composable () -> Unit,
)
